### PR TITLE
#166214085 integrate coveralls code climate

### PR DESCRIPTION
**Pull request by Nennyfills**
### Description
Integrate coveralls and codeclimate code coverage service

#### Type of change
- Install coveralls
- Add coveralls badge to readme.md file
- Add codeclimate badge to readme.md file
- Add coveralls-coveraget script
- Add coverage script

#### How Can This Been Tested?
N/A

#### Checklist:
N/A

#### Pivotal story
[#166214085](https://www.pivotaltracker.com/story/show/166214085)

#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by Lundii — CHANGES_REQUESTED** [review]
I tested it out. 

* first, you have 4 commits
* your pull request title does not follow the convention
* I ran `npm run coverage` after everything ran, I got the error below. Why is that 
image

**Comment by Nennyfills** [comment]
> I tested it out.
> 
> * first, you have 4 commits
> * your pull request title does not follow the convention
> * I ran `npm run coverage` after everything ran, I got the error below. Why is that
>   image

t

> I tested it out.
> 
> * first, you have 4 commits
> * your pull request title does not follow the convention
> * I ran `npm run coverage` after everything ran, I got the error below. Why is that
>   image
have fixed the commits
have corrected the PR title
for this to run I need a Travis.yml file, someone else is working on Travis branch.

**Review by Adekoreday — CHANGES_REQUESTED** [review]
kindly rebase this branch so we can proceed

**Review by Lundii — COMMENTED** [review]
The travis branch have been merged. Please do a rebase

**Review by allebd — CHANGES_REQUESTED** [review]
Awaiting a rebase

**Comment by Nennyfills** [comment]
> kindly rebase this branch so we can proceed

done

**Comment by Nennyfills** [comment]
> The travis branch have been merged. Please do a rebase

done

**Comment by Lundii** [comment]
> > The travis branch have been merged. Please do a rebase
> 
> done

i checked the migration folder of the develop branch (inside database folder). It contains some files your branch doesn't have. Can you please check it out to confirm, because a PR was merged recently.

**Review by Adekoreday — CHANGES_REQUESTED** [review]
you need to rebase again your server files are different from one on develop

File: coverall.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
+repo_token: UlARnqYpnmnGghoCE6IDSshcrqo1uPJ0z
```
Comment (allebd) [inline coverall.yml]: New line required

File: coverals.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
```
Comment (allebd) [inline coverals.yml]: Your file is name coverals.yml instead of .coveralls.yml

File: package.json [hunk 3]
```diff
@@ -9,13 +9,12 @@
...
         "heroku-postbuild": "npm run build",
         "test": "nyc mocha --timeout 10000 -r esm --exit",
-        "generate-lcov": "nyc report --reporter=text-lcov > lcov.info",
         "db:rollback": "npm run db:back && npm run db:roll && npm run db:seed",
         "db:roll": "sequelize db:migrate",
         "db:back": "sequelize db:migrate:undo:all",
         "db:seed": "sequelize db:seed:all",
-        "coverage": "npm run generate-lcov"
-    },
+        "coverage": "nyc report --reporter=text-lcov | coveralls",
```
Comment (allebd) [inline package.json]: You have a comma at the end of this line which is not meant to be there

File: package.json [hunk 4]
```diff
@@ -9,13 +9,12 @@
...
+},
```
Comment (allebd) [inline package.json]: The closing bracket is wrongly positioned

**Review by allebd — DISMISSED** [review]

**Review by henryade — COMMENTED** [review]
repair your commit message and i will approve it
here is my commit message, please high light the wrong part.
ch(integrate)coveralls and codeclimate 
- coveralls
- add coveralls badge to readme.md file
- add codeclimate badge to readme.md file
- add coveralls-coveraget script
- add coverage script
[#166214085]

**Comment by Lundii** [comment]
> repair your commit message and i will approve it

Nenny rewrite you commit to follow the convention.

File: package.json (hunk 4 above)
Comment (Adekoreday) [reply package.json]: the script to generate the lcov file will be used by coverage and codeclimate

**Review by Adekoreday — CHANGES_REQUESTED** [review]
i think you need to handle code coverage reporting script as well 
i can only see coveralls
a better approach might to put the previous generate lcov script back and use the script below

kindly explain any other goals concerning this task

"coveralls-coverage": "coveralls < lcov.info",
    "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",

for code climate and coveralls 
that script you inserted is only reporting to coveralls

you might install coveralls and codeclimate test reporter package

**Comment by Nennyfills** [comment]
> i think you need to handle code coverage reporting script as well
> i can only see coveralls
> a better approach might to put the previous generate lcov script back and use the script below
> kindly explain any other goal behind this task

> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",
> 
> for code climate and coveralls
> that script you inserted is only reporting to coveralls

please, can you research on this?
> "coverage": "nyc report --reporter=text-lcov | coveralls" is also equal 
> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",

ok will do

**Comment by Nennyfills** [comment]
> i think you need to handle code coverage reporting script as well
> i can only see coveralls
> a better approach might to put the previous generate lcov script back and use the script below
> 
> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",
> 
> for code climate and coveralls
> that script you inserted is only reporting to coveralls
> 
> you might install coveralls and codeclimate test reporter package
We are already using coveralls to report test coverage, please can you explain to me why we need code climate text coverage as well, meaning you want us to have two test coverage

**Comment by Nennyfills** [comment]
> > repair your commit message and i will approve it
> 
> Nenny rewrite you commit to follow the convention.

here is my commit message, please high light the wrong part.
ch(integrate)coveralls and codeclimate
coveralls
add coveralls badge to readme.md file
add codeclimate badge to readme.md file
add coveralls-coveraget script
add coverage script
[#166214085]

ok you mean your PR 
I can relate with this ok
will proceed to test your work locally

**Review by Lundii — APPROVED** [review]
Great

**Review by henryade — APPROVED** [review]

**Review by Adekoreday — APPROVED** [review]

**Review by KvNGCzA — APPROVED** [review]

**Pull request merged**